modules/smitten/index.ts

```diff
@@ -6,24 +6,35 @@ import {action} from '@action-land/core'
 
 export interface Smitten<T extends string | number = string | number> {
   of<S extends string | number>(type: T): Smitten<S>
-  emit(obj: any): void
+  emit(value: any): void
+  bindTo<S extends string | number>(value: any): Smitten<S>
```

Comment:
bindTo should return `Smitten<T>`

Reply:
Can we rename it to `emitWith`?

Reply:
How about just naming it `with`?

So it could be used as: `e.emit('action').with('payload')` 
- which reads like a natural sentence only, imo.

Otherwise it would be: `e.emit('action').emitWith('payload')`